code/modules/vending/_vending.dm

```diff
@@ -282,12 +282,11 @@
 			var/datum/data/vending_product/R = record
 
 			//first dump any of the items that have been returned, in case they contain the nuke disk or something
-			if(LAZYLEN(R.returned_products))
-				for(var/obj/returned_obj_to_dump in R.returned_products)
-					R.returned_products -= returned_obj_to_dump
-					returned_obj_to_dump.forceMove(get_turf(src))
-					step(returned_obj_to_dump, pick(GLOB.alldirs))
-					R.amount--
+			for(var/obj/returned_obj_to_dump in R.returned_products)
+				R.returned_products -= returned_obj_to_dump
```

Comment:
```suggestion
				LAZYREMOVE(R.returned_products, returned_obj_to_dump)
```
To make this into a lazylist, replace any insertions to the list with `LAZYADD(R.returned_products, inserted_item)`, and removals with `LAZYREMOVE(R.returned_products, removed_item)`, and any times you're accessing a specific index like `R.returned_products[i]` with `LAZYACCESS(R.returned_products, i)`.

This will make it so the lists is created when things are added to it, and nulled out when it's empty, saving memory.